llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -1687,6 +1687,44 @@ static Value *foldSelectInstWithICmpConst(SelectInst &SI, ICmpInst *ICI,
   return nullptr;
 }
 
+static Value *foldSelectInstWithICmpOr(SelectInst &SI, ICmpInst *ICI,
+                                       InstCombiner::BuilderTy &Builder) {
+
+  // a > -1 ? 1 : (a | value) --> smin(1, a | value)
+  // a >= 0 ? 1 : (a | value) --> smin(1, a | value)
+  const APInt *Cmp;
+  Value *A = ICI->getOperand(0);
+  Value *B = ICI->getOperand(1);
+  Type *Ty = SI.getType();
+  Value *TVal = SI.getTrueValue();
+  Value *FVal = SI.getFalseValue();
+  Constant *One = ConstantInt::get(Ty, 1);
+  Constant *NegOne = ConstantInt::get(Ty, -1);
+  Constant *Zero = Constant::getNullValue(Ty);
+  CmpInst::Predicate Pred = ICI->getPredicate();
+
+  if (!match(B, m_APIntAllowUndef(Cmp)))
```

Comment:
If you are going to allow undef, please add a vector test with some undef elements.

Reply:
But either way, this should just be `if(!match(B, m_AllOnes())`